CMakeLists.txt

```diff
@@ -79,28 +79,34 @@ if(NOT APPLE_IOS AND NOT ANDROID)
 	option(ENABLE_MONOLITHIC_INSTALL "Install everything in single directory on Linux and Mac" OFF)
 endif()
 
-# On Linux, use -DCMAKE_CXX_COMPILER_LAUNCHER=ccache instead.
-# The XCode and MSVC builds each require some more configuration, which is enabled by the following option:
-if(MSVC OR (CMAKE_GENERATOR STREQUAL "Xcode"))
-  option(ENABLE_CCACHE "Speed up recompilation by caching previous compilations" OFF)
+option(ENABLE_CCACHE "Speed up recompilation by caching previous compilations" OFF)
+if(ENABLE_CCACHE)
+	find_program(CCACHE ccache)
+	if(CCACHE-NOTFOUND)
+		message(FATAL_ERROR "'ccache' could not be found; install it or set ENABLE_CCACHE=OFF.")
```

Comment:
If you really want to do it, just add REQUIRED to find_program

Reply:
in my view, if you're passing ENABLE_CCACHE=ON explicitly, you know what you're doing and what you're expecting to happen :)

Reply:
You don't necessarily pass it explicitly, you could be using a preset that passes it. Currently all such presets are explicitly suffixed with `_ccache` so my argument is not very strong obviously :), but I don't see any problems with keeping it the way it is - do you?

Reply:
+1

Why reinvent the wheel? What's so special about ccache needs separate handling?

Let's assume that CMake error reporting is bad - why we only produce "nice" error message for ccache? Why we are not using same approach for our library dependencies?

Reply:
I think any dependency that is enabled by a cmake option should refer to that option in case of error. It's very little risk involved with having an error message with helpful info. I also disagree with the sentiment that it's reinventing the wheel if a custom error message contains more information than the default.

Regardless, I'm new on the project and I value confirming to project standards e.g using REQUIRED in favor of custom error messages, I'll make a new commit with `REQUIRED` instead :)

Reply:
In that case - upstream bug. Please send report to CMake team.

No need to add custom behavior that we need to support. I also don't see any other project similar to vcmi that uses something other than REQUIRED flag to report missing libraries.

Reply:
> In that case - upstream bug. Please send report to CMake team.

Not sure what you're referring to. @Kuxe means that it's kind of better to show custom message if the find error depends on a project option, I don't see any cmake bug here.

Reply:
What I am saying is that "show human readable message if required library is not found" should not be something we do on our side and is problem with CMake.

Likely not the one that they will even acknowledge, sure, but not something that should be done on our side.

Reply:
But CMake can't possibly know if a `find_program` is guarded by an option. The best CMake can do is just to tell that the program couldn't be found, but we shouldn't expect CMake to be able to infer what option was used to guard the `find_program` call to begin with.

```
if(ENABLE_<myprogram>)
  // What user-feedback can we provide if  this fails?
  // Could just fallback to telling user that <myprogram> wasn't found (default behaviour and current behavior as of 25be15a)
  // or we could say that <myprogram> wasn't found plus mention that it tried to be found to begin with due to setting ENABLE_<myprogram> to 'on'
  // regardless this is not something that CMake would be able to deduce on it's own
  // unless there was some "find_program_if"...
  find_program(MYPROGRAM myprogram REQUIRED)
endif()